### ISSUE TITLE
Linux 'sh' shell command output not shown on Linux

### ISSUE THREAD

**Issue by Orfheo** [Fish-Git/hyperion#95]
In the standard hercules-3.13 the output goes to the console, while with the current master version of hyperion I can't get any output.

Command gets executed (a simple "sh > /tmp/test" create the "test" output), but I can't see any output on the console.

Comparing the hyperion code with the hercules-3.13 current code, I found that, in logger.c, the hyperion:logger_thread() doesn't dup2() STDOUT_FILENO on logger_syslogfd[LOG_WRITE].

hercules-3.13:logger.c does, just at the beginning of logger_thread():

```C
#if !defined( _MSVC_ )
    /* Redirect stdout to the logger */
    if(dup2(logger_syslogfd[LOG_WRITE],STDOUT_FILENO) == -1)
    {
        if(logger_hrdcpy)
            fprintf(logger_hrdcpy, _("HHCLG001E Error redirecting stdout: %s\n"), strerror(errno));
        exit(1);
#endif /* !defined( _MSVC_ ) */
```
while hyperion doesn't, for what I can see grepping sources.

I did some test adding back this fragment of code just after the lines:

```C
    /* ZZ FIXME:  We must empty the read pipe before we terminate */
    /* (Couldn't we just loop waiting for a 'select(,&readset,,,timeout)'
        to return zero?? Or use the 'poll' function similarly?? - Fish) */
```
and I got the shell commands output back.

I've a git patch tested and ready, in the case you want to just apply it.

My nick, Orfheo, belong to "Peppe" (Giuseppe Vitillaro) on the main hercules group, where I was asked to open this issue.

Regards.
Peppe.

**Comment by Fish-Git** [comment]
FYI: I originally removed the `dup2` code for Linux to resolve and issue where duplicate messages were being issued during shutdown _(commit c0074177fddbc0e13003cbc2a0c808b34fff5ef3 "Fix duplicated messages at shutdown on Linux")_ but did not realize doing so would cause the undesirable behavior you describe. My apologies!

I will research this issue further to try and determine a better way to to prevent duplicate messages from ocurring at shutdown on Linux _without_ also preventing shell commands from being displayed.

It might take me a while to eventually get this fixed _(I'm working on many things at the same time!)_ so your patience is appreciated. (But I _will_ get this fixed for you!)

Thank you for reporting this, Peppe.

**Comment by Fish-Git** [comment]
Resolved by commit 48cc73b685f91486029a45b1e4da76d389f2e3ab.

**Comment by Orfheo** [comment]
Merged to my personal branch without problems, the shell commands output is back on the console and, at exit, messages are not repeated. Nice fix!
